src/Data/Poly/Internal/Dense.hs

```diff
@@ -142,6 +145,11 @@ leading (Poly v)
   | G.null v  = Nothing
   | otherwise = Just (fromIntegral (G.length v - 1), G.last v)
 
+degree :: G.Vector v a => Poly v a -> Word
+degree p = case leading p of
+  Nothing     -> 0
```

Comment:
This is wrong, `degree 0 = -∞` and not `0`. Since you cannot represent `-∞ :: Word`, it would be better to return `Maybe Word` by defining `degree = fmap fst . leading`.